src/main/java/org/aksw/simba/lemming/metrics/single/AvgVertexDegreeMetric.java

```diff
@@ -1,37 +1,151 @@
+
+    public AvgVertexDegreeMetric() {
+        super("avgDegree");
+    }
+
+    protected AvgVertexDegreeMetric(String name) {
+        super(name);
+    }
+
+    @Override
+    public double apply(ColouredGraph graph) {
+        return calculateAvg(graph.getGraph().getAllInEdgeDegrees());
+    }
+
+    protected double calculateAvg(IntArrayList degrees) {
+        double sum = 0;
+        for (int i = 0; i < degrees.size(); ++i) {
+            sum += degrees.getInt(i);
+        }
+        return sum / degrees.size();
+    }
+    
+	/**
+	 * Stores the previously computed values in UpdateMetricResult object
+	 * 
+	 * @param graph
+	 *            - input graph.
+	 * @param previousResult
+	 *            - UpdatableMetricResult object containing the previous results
+	 *            that can be used.
+	 * @return UpdatableMetricResult object with updated sum that can be used in
+	 *         further iterations.
+	 */
+	@Override
+	public UpdatableMetricResult applyUpdatable(ColouredGraph graph, UpdatableMetricResult previousResult) {
+
+		AvgVertexDegreeMetricResult metricResultTempObj = new AvgVertexDegreeMetricResult(getName(), 0.0);
+		if (previousResult instanceof AvgVertexDegreeMetricResult) {
+			// Copying previously computed values in temporary variables
+			metricResultTempObj.setSumVertexDeg(((AvgVertexDegreeMetricResult) previousResult).getSumVertexDeg());
+			metricResultTempObj
+					.setNumberOfVertices(((AvgVertexDegreeMetricResult) previousResult).getNumberOfVertices());
+		}
+
+		return metricResultTempObj;
+	}
+
+	/**
+	 * The method contains logic that computes the average vertex degree metric
+	 * efficiently. If the metric is computed for the first time then it uses the
+	 * degrees stored in VertexDegrees else it will update the previously stored sum
+	 * value.
+	 * 
+	 * @param triple
+	 *            - edge on which graph operation is performed.
+	 * @param metric
+	 *            - input metric which needs to be computed.
+	 * @param graph
+	 *            - input graph.
+	 * @param graphOperation
+	 *            - boolean value indicating graph operation. ("true" for adding an
+	 *            edge and "false" for removing an edge)
+	 * @param previousResult
+	 *            - UpdatableMetricResult object containing the previous computed
+	 *            results.
+	 * @return UpdatableMetricResult object with updated values that can be used in
+	 *         further computations
+	 */
+	@Override
+	public UpdatableMetricResult update(TripleBaseSingleID triple, ColouredGraph graph, boolean graphOperation,
+			UpdatableMetricResult previousResult, VertexDegrees mVertexDegrees) {
+
+		if (getName().equals("stdDevInDegree") || getName().equals("stdDevOutDegree")) {
+			return new SimpleMetricResult(getName(), apply(graph));
+		}
+
+		AvgVertexDegreeMetricResult metricResultTempObj = (AvgVertexDegreeMetricResult) applyUpdatable(graph, previousResult);
```

Comment:
There is a general misunderstanding of the meaning of the two newly added methods. Calling applyUpdatable within the update method does not really make sense to me. Please check again the description of the methods.

Reply:
okay. I have removed the call to applyUpdatable method, using the previousResult object to get the old computed values.